### PR TITLE
refactor isTv

### PR DESCRIPTION
**Pull request by thonsi**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [*] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Prepare way to detect the TV 

#### APK testing
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

### CODE REVIEW

**Comment by Redirion** [comment]
so the functional difference is a null check. Is there any device that happens to not have UiModeManager? Seems to be a core service always launched by SystemServer. I don't see any advantages of this refactoring. Sorry

File: app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java [hunk 1]
```diff
@@ -57,8 +58,11 @@ public static boolean isTv(final Context context) {
         final PackageManager pm = App.getApp().getPackageManager();
 
         // from doc: https://developer.android.com/training/tv/start/hardware.html#runtime-check
-        boolean isTv = ContextCompat.getSystemService(context, UiModeManager.class)
-                .getCurrentModeType() == Configuration.UI_MODE_TYPE_TELEVISION
+        @Nullable
+        UiModeManager uiModeManager =
+                (UiModeManager) context.getApplicationContext().getSystemService(UI_MODE_SERVICE);
```
Comment (snaik20) [inline app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java:63]: Any reason to prefer `UI_MODE_SERVICE`. Please provide a reason or a doc reference to cross verify this. Also, provide a test case that returns `null` for existing function as pointed in another comment.
